intermediate-js.md

```diff
@@ -3,37 +3,42 @@
 ### Without using Google answer the following:
 
 1. What is the difference between .map() and .filter()?
-
+.map adds something to an array, .filter takes something out of an array
 2. Why would you use object destructuring?
-
+It is easier to use a destructured object when calling on it in a function than a full object path.
 3. What is the difference between var, let, and const?
-
+var is a variable that can be changed, const is a variable that is constant or cannot be changed, let is a variable used in for loops and can be changed.
 4. Why is testing important?
-
+Testing shows you whether your functions/codes work and if they don't, where the mistakes in the coding are.
 5. What is a higher order function?
-
+a function that calls another function or object.
```

Comment:
Great! You've described one kind of higher order function. What is the other way that something can be a higher order function?